crates/circuit/src/packed_instruction.rs

```diff
@@ -614,7 +614,7 @@ pub struct PackedInstruction {
     /// which is a simple null-pointer check.
     ///
     /// WARNING: remember that `OnceLock`'s `get_or_init` method is no-reentrant, so the initialiser
-    /// must not yield the GIL to Python space.  We avoid using `GILOnceCell` here because it
+    /// must not yield the GIL to Python space.  We avoid using `PyOnceLock` here because it
```

Comment:
Is this statement true now the sync primitive is `PyOnceLock` rather than `GILOnceCell`, including the bit about `Clone` below?

Reply:
I think it's still true `PyOnceLock::get()` requires the py token: https://pyo3.rs/main/doc/pyo3/sync/struct.pyoncelock#method.get

Reply:
That being said I did just obviously find/replace all the `GILOnceCell` instances in the codebase with `PyOnceLock` to write this PR and this was just caught up in that. I can revert this change if you'd prefer, I don't think we should change the synchronization type here in this PR even if we could.

Reply:
It's more that I don't know if avoiding `GILOnceCell` makes sense any more - it might be better just to use `PyOnceLock`? We can leave it for now though - I have a vague memory that part of the comment already got obsoleted in a different change. (I'm on my phone and it's hard to check properly)